### ISSUE TITLE
Abundance estimation after Minimap2 alignment 

### ISSUE THREAD

**Issue by foolsgold** [gpertea/stringtie#207]
Hi Geo,

I have nanopore cDNA data aligned with minimap2 (with --cs --MD). I tried abundance estimation with StringTie v1.3.5 that should recognize the "CS" tag as an alternative to "XS". The spliced reads are identified but I find that abundance estimates are not calculated correctly. I have attached an example region to show my issue. There is a screenshot of Minimap2 (ONT) and Hisat2 (Illumina) reads alignment at the HAMP locus, and stringtie quantification of the Minimap2 alignment bams. The read alignments show highest abundance isoform as the transcript with 3 exons (ENST00000222304). This is also identified as the highest abundance transcript with Hisat2 alignment (stringtie counts not attached). However in the Minimap2 quantification with stringtie (t_data.ctab snippet in screenshot), the isoform with 4 exons (ENST00000598398) appears to have maximum abundance while the other isoforms have extremely low abundance. Do we need to change some parameters for abundance estimation (as well as assembly) with long spliced reads aligned with minimap2?
I'd appreciate your advice.
Thanks!

<img width="974" alt="screen shot 2018-11-20 at 14 14 50" src="https://user-images.githubusercontent.com/8257635/48779088-b6daa280-ecce-11e8-849e-d9b33c057954.png">

![screen shot 2018-11-20 at 14 11 36](https://user-images.githubusercontent.com/8257635/48778894-4a5fa380-ecce-11e8-9279-908290bedd02.png)

**Comment by gpertea** [comment]
At this stage we don't have much experience with using solely minimap2 alignments for abundance estimation with the current version of StringTie. We added that "cs" tag support as a simple technical convenience, only testing it with minimap2 alignments in very simple cases. Hence I cannot provide any advice here as I am as puzzled as you are at your observations here in this example, so I'd like to take a closer look if possible.
I am trying to make sense of the figure you attached and admittedly I cannot even see the 4th exon on the left there (which is the 1st exon for that 4-exon isoform of course) being represented in the ONT alignments (or any alignments in that picture, actually). Also, probably unimportant but I'm not sure why there are both blue and red junction arcs there for ONT alignments, while for the Illumina ones there is only the red arc junction coloring - what does the blue coloring mean again, are those junctions found on the opposite strand? (sorry I haven't used IGV with junction display enabled).
I am wondering if you can share some of the minimap2 alignments for this example region so I can take a closer look at how the 4-exon isoform ends up with such a high coverage..

**Comment by foolsgold** [comment]
Thanks a lot for the prompt reply! 
> I am trying to make sense of the figure you attached and admittedly I cannot even see the 4th exon on the left there (which is the 1st exon for that 4-exon isoform of course) being represented in the ONT alignments (or any alignments in that picture, actually).
Yes, there are a few reads that align to the 1st exon but most reads align from the 2nd exon only.

> Also, probably unimportant but I'm not sure why there are both blue and red junction arcs there for ONT alignments, while for the Illumina ones there is only the red arc junction coloring - what does the blue coloring mean again, are those junctions found on the opposite strand? (sorry I haven't used IGV with junction display enabled).
The blue colouring is supposed to represent junctions on the (-) strand. However, here is just reflects the unstranded data. As the default nanopore protocol is unstranded, my single read fastq file does not contain strand information and I have reads aligning to both strands. The orientation of the read is inferred and written to the file at ts: SAM tag. 

> I am wondering if you can share some of the minimap2 alignments for this example region so I can take a closer look at how the 4-exon isoform ends up with such a high coverage..
 I have uploaded the sam file from the region here - 
https://www.dropbox.com/s/vf7uqjf0jn8iz9x/HAMP.sam?dl=0
Let me know if you are able to access it and if it is useful. 
Thanks a lot for your help!

**Comment by foolsgold** [comment]
Thank you! Sorry if I added to the confusion regarding the tags. There is also an optional "cs" SAM/PAF tag in minimap2 output - this encodes the alignment, mismatches, INDELs, splice signal - similar to the MD tag.

**Comment by foolsgold** [comment]
Yes, I found this in the abundance estimation with known isoforms steps. I did not try the assembly. I have uploaded the GTF file containing just these reference isoforms [here](https://www.dropbox.com/s/sxls6f8y7af37m0/HAMP.gtf?dl=0) .

**Comment by foolsgold** [comment]
I have repeated the abundance estimation without restricting to known isoforms and find the transcript with three exons (ENST00000222304) as the most abundant transcript in all samples, both nanopore and Illumina. Moreover, the expression of other isoforms is extremely low, as observed in the actual alignment as well.    


### ONT![ONT](https://user-images.githubusercontent.com/8257635/48803734-b3fda300-ed0a-11e8-946d-f3055c6ed435.png)

### Illumina![Illumina](https://user-images.githubusercontent.com/8257635/48804131-b7455e80-ed0b-11e8-97e1-eec9b0a48e8e.png)

**Comment by foolsgold** [comment]
Hi Geo,
Do you have any additional thoughts on why I see better abundance estimate without the "-e " option. From the manual I actually assumed the contrary, i.e. abundance estimation with "-e" should be more accurate when assembly is not required.
Thanks!
